### DISCUSSION TITLE
Is there a way to check if the TCP-client is connected to the server ?

### DISCUSSION THREAD

**Discussion by KLelong** [eModbus/eModbus#354] [Q&A]
When I code for MQTT  I first try to see if the server can be reached. Then I publish topics.
With the emodbus-library, I seem to have to send a real request to the server before I can see if the server is reachable and then use the result of the ErrorHandler. 
Or is there a way to check before sending a request ?

**Comment by Miq1** [comment]
> Or is there a way to check before sending a request ?

Not that I would be aware of it. The Modbus protocol has no notion of a "ping" request or such. The only thing to detect if a Modbus server is up and running is to send a proper request - and that is what is done anyway.

**Comment by KLelong** [comment]
I had hoped that successfully opening the TCP-socket could be detected by the library.
Maybe I should test this myself.

**Reply by Miq1** [reply]
In fact it can be detected, but that currently will happen only when a request is about to be sent. Sorry... 😉
